### DISCUSSION TITLE
How to align center align a block of text

### DISCUSSION THREAD

**Discussion by kompre** [quarto-dev/quarto-cli#10698] [Q&A] [answered]
### Description

What is the most quarto way to center align a block of text like in the below example? 

````qmd

---
title: "How to align text in div?"
format: pdf

---

we the people


```{=latex}
\begin{center}
    DECLARE!
\end{center}
```


we are pretty cool

````

Ideally I would like to avoid the raw latex block if possible, also because it does not work for html output; I would rather enclose the text in `:::`, but i could not figure it out. I tried something like this 

````qmd

::: {layout="[-1,1,-1]" layout-align="center"}

DECLARE!

:::

````

but the text is rendered as left aligned in the middle block (as expected really). the option `layout-align="center"` does not seems to change anything

**Comment by mcanouil** [comment]
There is no "markdown" way.
Don't mix "position" and "justification/alignment". Layout is for position.

See:

- https://github.com/jgm/pandoc/issues/719

**Reply by kompre** [reply, marked as answer]
I stumbled upon this [custom blocks in R Markdown](https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html#block-syntax) that explain how to convert a div block in a any latex environment by adding `data-latex=""` as parameter. 

For example my initial example could be written as:

```md
---
title: "How to align text in div?"
format: pdf

---

we the people


::: {.center data-latex=""}

DECLARE!

:::

we are pretty cool
```

<details>
<summary>tex output</summary>

The div get rendered as follows in the `tex`

```latex
\begin{center}
DECLARE!

\end{center}
```


</details>



This should work also for `html` output with by either adding `style="text-align: center;"` or by defining a custom css like so:

```css
.center {
    text-align: center;
}
```

As a very welcome bonus the content inside the div could be treated as regular quarto markdown (bolded, italics, shortcode, ...)

I prefer this approach because I feel is more in the spirit of quarto, and I don't have to duplicate content for the sake of different output.

I don't think I have found any mention of this in the official documentation, or at least in the [pdf basics section](https://quarto.org/docs/output-formats/pdf-basics.html) which I am inclined to consult when to do stuff in pdf. Maybe would be worth adding a note to it?

**Reply by mcanouil** [reply]
That's Pandoc divs, see the documentation: https://quarto.org/docs/authoring/markdown-basics.html#sec-divs-and-spans

Note that if you want a format agnostic approach, you can write a LUA filter to turn a div with a specific attribute into the right code for the format.
